tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -34,30 +37,34 @@ def __init__(self,
                              *args,
                              **kwargs)
         self.test_ctx = test_ctx
-        self.filesystem_catalog_mode = filesystem_catalog_mode
-        self.spark: Optional[SparkService] = None
-        self.trino: Optional[TrinoService] = None
+        self.catalog_service.set_filesystem_wrapper_mode(
+            filesystem_catalog_mode)
+        self.included_query_engines = include_query_engines
+        # To be populated later once we have the URI of the catalog
+        # available
+        self.query_engines = []
 
     def setUp(self):
-        self.catalog_service.set_filesystem_wrapper_mode(
-            self.filesystem_catalog_mode)
         super().setUp()
-        self.spark = SparkService(self.test_ctx,
-                                  str(self.catalog_service.catalog_url),
-                                  self.redpanda.si_settings)
-        self.spark.start()
-        self.trino = TrinoService(self.test_ctx,
-                                  str(self.catalog_service.catalog_url),
-                                  self.redpanda.si_settings)
-        self.trino.start()
+        for engine in self.included_query_engines:
+            svc_cls = get_query_engine_by_type(engine)
+            svc = svc_cls(self.test_ctx, str(self.catalog_service.catalog_url),
+                          self.redpanda.si_settings)
+            svc.start()
+            self.query_engines.append(svc)
```

Comment:
nit: maybe make this a map? That way a test can operate on a given service if it contains multiple engines

Reply:
ya I was hoping to add a method get_engine_by_type() but nothing is really using it. Also I think we can just loop through it to find based on `engine_name()`